File: encrypt.rb
#!/usr/bin/env ruby

require "base64"

def crypto(string, rounds, method)
    rounds.times do |i|
        if method == "encrypt"
            string = Base64.encode64(string)
        elsif method == "decrypt"
            string = Base64.decode64(string)
        end
    end
    return string
end

string = ARGV[0]
rounds = Integer(ARGV[1])

enc_value = crypto(string, rounds, "encrypt")

puts "\"encrypted\" #{enc_value}"
puts "\"decrypted\" #{crypto(enc_value, rounds, "decrypt")}"

File: encrypt.js
#!/usr/bin/env node

const crypto = (string, rounds, method) => {
    for (let i = 0; i < rounds; i++) {
        if (method == 'encrypt') {
            string = Buffer.from(string).toString('base64');
        } else if (method == "decrypt") {
            string = Buffer.from(string, 'base64').toString('ascii');
        }
    }
    return string;
}

string = process.argv[2];
rounds = process.argv[3];

let encvalue = crypto(string, rounds, "encrypt");

console.log(`"encrypted" ${encvalue}"`);
console.log(`"decrypted" ${crypto(encvalue, rounds, "decrypt")}`);
